src/frontend/mame/mame.cpp

```diff
@@ -318,10 +318,18 @@ void emulator_info::display_ui_chooser(running_machine& machine)
 		ui::menu_select_game::force_game_select(mui, container);
 }
 
-int emulator_info::start_frontend(emu_options &options, osd_interface &osd, int argc, char *argv[])
+int emulator_info::start_frontend(emu_options &options, osd_interface &osd, std::vector<std::string> &args)
 {
 	cli_frontend frontend(options, osd);
-	return frontend.execute(argc, argv);
+	return frontend.execute(args);
+}
+
+int emulator_info::start_frontend(emu_options &options, osd_interface &osd, int argc, char *argv[])
+{
+	std::vector<std::string> args(argc);
```

Comment:
Why not use range-based construction and skip the loop?

    std::vector<std::string> args(argv, argv + argc);